classes/Cart.php

```diff
@@ -1405,7 +1405,7 @@ public function getProductQuantity($idProduct, $idProductAttribute = 0, $idCusto
         $parentSql = 'SELECT
             COALESCE(SUM(first_level_quantity) + SUM(pack_quantity), 0) as deep_quantity,
             COALESCE(SUM(first_level_quantity), 0) as quantity
-          FROM (' . $firstUnionSql . ' UNION ' . $secondUnionSql . ') as q';
+          FROM (' . $firstUnionSql . ' UNION ALL ' . $secondUnionSql . ') as q';
```

Comment:
Thanks for your contribution @tom-combet  ! Is this really necessary ? I fail to see the point. I don't know much about cart rules yet. so Maybe I just miss something :) Otherwise the PR sounds legit to me :) 👍

Reply:
Yep, because without it, duplicated results are ignored. Then if you have multiple customized rows of same product/combination and in same quantity, only one row of them will be counted in the sum.

By the way, it doesn't concern cart rules at all: it's just about cart products quantities here ;)